webapp/src/ts/modules/reports/reports-sidebar-filter.component.html

```diff
@@ -0,0 +1,119 @@
+<section class="sidebar-filter">
+  <div class="sidebar-backdrop" [ngClass]="{ hidden: !isOpen }" (click)="toggleSidebarFilter(false)"></div>
+
+  <div class="sidebar-main" [ngClass]="{ hidden: !isOpen }">
+    <div class="sidebar-header">
+      <p class="sidebar-title">{{ 'reports.sidebar.filter.title' | translate }}</p>
+      <a class="sidebar-close fa fa-times" (click)="toggleSidebarFilter(false)"></a>
+    </div>
+
+    <div class="sidebar-body">
+      <accordion [closeOthers]="true" [isAnimated]="true">
+        <!-- 1. Form Type -->
+        <accordion-group id="form-filter-accordion">
+          <ng-container
+            accordion-heading
+            [ngTemplateOutlet]="headerTemplate"
+            [ngTemplateOutletContext]="{
+              numSelected: filterCount.formFilter,
+              label: 'reports.sidebar.filter.form_type',
+              filters: [ 'formFilter' ]
+            }">
+          </ng-container>
+          <mm-form-type-filter
+            class="filter"
+            fieldId="formFilter"
+            (search)="applyFilters()"
+            [disabled]="disabled"
+            [inline]="true">
+          </mm-form-type-filter>
+        </accordion-group>
+
+        <!-- 2. Place -->
+        <accordion-group id="place-filter-accordion">
+          <ng-container
+            accordion-heading
+            [ngTemplateOutlet]="headerTemplate"
+            [ngTemplateOutletContext]="{
+              numSelected: filterCount.placeFilter,
+              label: 'reports.sidebar.filter.place',
+              filters: [ 'placeFilter' ]
+            }">
+          </ng-container>
+          <mm-facility-filter
+            class="filter"
+            fieldId="placeFilter"
+            (search)="applyFilters()"
+            [disabled]="disabled"
+            [inline]="true">
+          </mm-facility-filter>
+        </accordion-group>
+
+        <!-- 3. Date -->
+        <accordion-group id="date-filter-accordion">
+          <ng-container
+            accordion-heading
+            [ngTemplateOutlet]="headerTemplate"
+            [ngTemplateOutletContext]="{
+              numSelected: (filterCount.fromDateFilter || 0) + (filterCount.toDateFilter || 0),
+              label: 'reports.sidebar.filter.date',
+              filters: [ 'fromDateFilter', 'toDateFilter' ]
+            }">
+          </ng-container>
+          <label>{{ 'reports.sidebar.filter.from_date' | translate }}</label>
+          <mm-date-filter
+            #fromDate
+            class="filter"
+            fieldId="fromDateFilter"
+            (search)="applyFilters()"
+            [disabled]="disabled"
+            [isStartDate]="true">
+          </mm-date-filter>
+
+          <label>{{ 'reports.sidebar.filter.to_date' | translate }}</label>
+          <mm-date-filter
```

Comment:
I expect there is an easy CSS fix for this, but I did notice that the date widget for the `to` date gets cutoff if the window is not tall enough.  (Cannot see the bottom row)

![image](https://user-images.githubusercontent.com/8441903/181632550-53838b51-b1e6-46c1-b24c-283c94a05a18.png)

Reply:
This dropdown calendar component is a floating element outside the scroll scope of the container (it won't scroll with the rest of things), its position is calculated with javascript. Other good libraries automatically recalculate and re-position the calendar when it can't fit, but the library that we're using is limited and doesn't do that.

I've added a workaround but it's not perfect. Tiny devices like probably some very old Tecno phone, won't fit when opening on top or below the date input.

I'm recommending to the team to move the "date filter" section on top of the "form filter" that way we ensure the calendar will fit in the screen, even if it's not the most used filter by the user.

The date component has been a challenge while working on this assignment, it has limited features for what we need in the new design, [apparently not supported anymore](https://github.com/dangrossman/daterangepicker/discussions/2266#discussioncomment-2594296), not interest of merging contributions ([55 PR](https://github.com/dangrossman/daterangepicker/pulls)), last release was 2 years ago but we can't upgrade ([Dont upgrade bootstrap-daterangepicker](https://docs.communityhealthtoolkit.org/core/guides/update-dependencies/#problems)) anyways. I quick look for a better option but the ones I found didn't supported Nepali out of the box, we need to look more and change it for a smarter modern option that supports all languages (I'll open a ticket for this)